tsl/src/compression/simple8b_rle_bitmap.h

```diff
@@ -0,0 +1,173 @@
+		const uint64 selector_mask = 0xFULL << selector_shift;
+		const uint8 selector_value = (slot_value & selector_mask) >> selector_shift;
+		Assert(selector_value < 16);
+
+		uint64 block_data = compressed_data[block_index];
+
+		if (simple8brle_selector_is_rle(selector_value))
+		{
+			/*
+			 * RLE block.
+			 */
+			const int32 n_block_values = simple8brle_rledata_repeatcount(block_data);
+			CheckCompressedData(n_block_values <= GLOBAL_MAX_ROWS_PER_COMPRESSION);
+
+			const uint64 repeated_value = simple8brle_rledata_value(block_data);
+			CheckCompressedData(repeated_value <= 1);
+
+			CheckCompressedData(decompressed_index + n_block_values <= num_elements);
+
+			for (int16 i = 0; i < n_block_values; i++)
+			{
+				bitmap_bools_[decompressed_index + i] = repeated_value;
+			}
+			decompressed_index += n_block_values;
+			Assert(decompressed_index <= num_elements);
+
+			num_ones += repeated_value * n_block_values;
+		}
+		else
+		{
+			/*
+			 * Bit-packed block. Since this is a bitmap, this block has 64 bits
+			 * packed. The last block might contain less than maximal possible
+			 * number of elements, but we have 64 bytes of padding on the right
+			 * so we don't care.
+			 */
+			CheckCompressedData(selector_value == 1);
+
+			Assert(SIMPLE8B_BIT_LENGTH[selector_value] == 1);
+			Assert(SIMPLE8B_NUM_ELEMENTS[selector_value] == 64);
+
+			/* Have to zero out the unused bits, so that the popcnt works properly. */
+			const int elements_this_block = Min(64, num_elements - decompressed_index);
+			Assert(elements_this_block <= 64);
+			/*
+			 * We should require at least one element from the block. Previous
+			 * blocks might have had incorrect lengths, so this is not an
+			 * assertion.
+			 */
+			CheckCompressedData(elements_this_block > 0);
+			block_data &= (-1ULL) >> (64 - elements_this_block);
+
+			/*
+			 * The number of block elements should fit within padding. Previous
+			 * blocks might have had incorrect lengths, so this is not an
+			 * assertion.
+			 */
+			CheckCompressedData(decompressed_index + 64 < num_elements_padded);
+
+#ifdef HAVE__BUILTIN_POPCOUNT
+			num_ones += __builtin_popcountll(block_data);
+#endif
+			for (int16 i = 0; i < 64; i++)
+			{
+				const uint64 value = (block_data >> i) & 1;
+				bitmap_bools_[decompressed_index + i] = value;
+#ifndef HAVE__BUILTIN_POPCOUNT
+				num_ones += value;
+#endif
+			}
+			decompressed_index += 64;
+		}
+	}
+
+	/*
+	 * We might have unpacked more because we work in full blocks, but at least
+	 * we shouldn't have unpacked less.
+	 */
+	CheckCompressedData(decompressed_index >= num_elements);
+	Assert(decompressed_index <= num_elements_padded);
+
+	/*
+	 * Might happen if we have stray ones in the higher unused bits of the last
+	 * block.
+	 */
+	CheckCompressedData(num_ones <= num_elements);
+
+	result.bitmap_bools_ = bitmap_bools_;
+	result.num_ones = num_ones;
+
+	/* Sanity check. */
+#ifndef NDEBUG
```

Comment:
Is there a reason for not using `USE_ASSERT_CHECKING` here?

Reply:
USE_ASSERT_CHECKING  is more suitable, right.